### ISSUE TITLE
Flagged cards in card browser distract more than they help

### ISSUE THREAD

**Issue by bnknndy** [ankidroid/Anki-Android#11671]
Flagged cards add noise and reduce legibility in the card browser.

Specifically, the card browser fills fields with flag colors, some of which cause the browser to violate [WCAG color contrast guidelines](https://webaim.org/articles/contrast/). Also, the fills may distract users when they try to read other cards.

![image](https://user-images.githubusercontent.com/7875524/174862306-4007f6ec-2896-45fa-adef-dcd5b592f011.png)

## Relevant heuristic

 [Aesthetic and minimalist design](https://www.nngroup.com/videos/aesthetic-and-minimalist-design) (*Video only. For a relevant article, consider supplementing with [Signal-to-Noise Ratio](https://www.nngroup.com/articles/signal-noise-ratio/).*)

"Remove unnecessary elements from the user interface and maximize the signal-to-noise ratio of the design."

## Student's severity classification (minor, major, critical)

*Major*, due to:

1. **Frequency**: "This is a common issue for users because the flagging system is a tool that a high percentage of users will utilize."

2. **Impact**: "The impact of this issue is moderate. The UI is more crowded with the colors taking up more of the screen and creating unnecessary noise that decreases the visibility of other aspects of the UI. This issue is not significant because flags are used sparingly by users but does negatively affects the user experience."

3. **Persistence**: "This issue has high persistence since the user will be affected by the issue whenever they are viewing their cards. This consistent encountering of the issue regardless of whether the user is aware of it makes the persistence high."

4. **Weighting**: "The frequency and persistence of this issue is high because the feature is commonly used and consistently something the user will encounter. The impact is moderate because it does not significantly affect the user and their experience. This persistence of the issue is high and results in a rating of 3 because of the consistent encountering of the issue."

This issues comes from [Source file](https://1sfu-my.sharepoint.com/personal/paulh_sfu_ca/_layouts/15/onedrive.aspx?id=%2Fpersonal%2Fpaulh%5Fsfu%5Fca%2FDocuments%2FCMPT%2D363%2D222%2Fassignments%2Fassignment%2D1%2Fankidroid%20flashcards%20team%2Fheuristic%2Deval%2Dexemplary%2D3%2Epdf&parent=%2Fpersonal%2Fpaulh%5Fsfu%5Fca%2FDocuments%2FCMPT%2D363%2D222%2Fassignments%2Fassignment%2D1%2Fankidroid%20flashcards%20team&ga=1) as discussed in #11664. *Text not wrapped in quotation marks is rewritten for brevity.*

**Comment by BrayanDSO** [comment]
Here I'd just choose colors with better contrast as Anki desktop does
![image](https://user-images.githubusercontent.com/69634269/175120726-d236220d-c398-4d05-ac91-e4c9d7bdf50b.png)

@viciousAegis is the one tweaking the card browser currently and may have something different in mind about flags

**Comment by viciousAegis** [comment]
The current color palette is:
<img width="1512" alt="Screenshot 2022-06-23 at 10 48 32 AM" src="https://user-images.githubusercontent.com/86671025/175220901-4b2d3179-1aaf-4f30-be7e-05945a6a4450.png">

I propose something like this, just muted shades of the colors (for the light theme). For the dark and black themes, we can use darker shades to contrast better with the text:
<img width="1507" alt="Screenshot 2022-06-23 at 10 48 16 AM" src="https://user-images.githubusercontent.com/86671025/175220979-75772f97-343c-49fa-8eb7-1f94a9096cb1.png">

## Before
![1655962500220](https://user-images.githubusercontent.com/86671025/175222927-4b722894-8edf-43d5-8af9-4ea59ba4e964.jpeg)

## After
![1655962500213](https://user-images.githubusercontent.com/86671025/175222953-19475f0a-9c3c-4346-9468-2e92a4810a09.jpeg)

**Comment by BrayanDSO** [comment]
As long as they pass accessibility scanner, I'm fine with any color choice.

**Comment by bnknndy** [comment]
@dorrin-sot I think that's a cool idea. Both proposed fixes have merit here, and might benefit from being combined.
 
Having some kind of flag icon displayed *and* more palatable background colors would provide helpful redundant signaling for everyone and extend accessibility for the colorblind.

**Comment by dorrin-sot** [comment]
Just realized that for the cards that are both flagged and marked we don't display them being marked (for example if a card is both red-flag and marked, it's just shown red). Would it be better to remove the coloring for marked cards completely and instead display the mark icon next to them? This would also fix the issue of the purple flag background being too similar to the marked background.
